elasticsearch/src/http/transport.rs

```diff
@@ -474,6 +519,102 @@ impl Transport {
         if let Some(q) = query_string {
             request_builder = request_builder.query(q);
         }
+        Ok(request_builder)
+    }
+
+    fn parse_to_url(address: &str, scheme: &str) -> Result<Url, Error> {
+        if address.is_empty() {
+            return Err(crate::error::lib("Bound Address is empty"));
+        }
+
+        let matches = ADDRESS_REGEX
+            .captures(address)
+            .ok_or_else(|| crate::lib(format!("error parsing address into url: {}", address)))?;
+
+        let host = matches
+            .name("fqdn")
+            .or_else(|| Some(matches.name("ip").unwrap()))
+            .unwrap()
+            .as_str()
+            .trim();
+        let port = matches.name("port").unwrap().as_str().trim();
+
+        Ok(Url::parse(
+            format!("{}://{}:{}", scheme, host, port).as_str(),
+        )?)
+    }
+
+    /// Creates an asynchronous request that can be awaited
+    pub async fn send<B, Q>(
+        &self,
+        method: Method,
+        path: &str,
+        headers: HeaderMap,
+        query_string: Option<&Q>,
+        body: Option<B>,
+        timeout: Option<Duration>,
+    ) -> Result<Response, Error>
+    where
+        B: Body,
+        Q: Serialize + ?Sized,
+    {
+        // Threads will execute against old connection pool during reseed
+        if self.conn_pool.reseedable() {
+            let local_conn_pool = self.conn_pool.clone();
+            let connection = local_conn_pool.next();
+
+            // Build node info request
+            let node_request = self.request_builder(
+                &connection,
+                Method::Get,
+                "_nodes/http?filter_path=nodes.*.http",
+                headers.clone(),
+                None::<&Q>,
+                None::<B>,
+                timeout,
+            )?;
+
+            spawn(move || {
```

Comment:
if we require tokio to be around anyways, wouldn't it be better (in terms of resource usage etc.) to spawn a tokio task for the reseeding?